zap/src/main/java/org/zaproxy/zap/control/AddOn.java

```diff
@@ -453,7 +453,11 @@ public static ValidationResult isValidAddOn(Path file) {
         try {
             return Optional.of(new AddOn(file));
         } catch (AddOn.InvalidAddOnException e) {
-            logger.warn("Invalid add-on: " + file.toString(), e);
+            if (logger.isDebugEnabled() || Constant.isDevMode()) {
+                logger.warn("Invalid add-on: " + file.toString(), e);
+            } else {
+                logger.warn("Invalid add-on: " + file.toString());
```

Comment:
Worth including the exception's message in this case. The common message could be extracted to avoid duplication.